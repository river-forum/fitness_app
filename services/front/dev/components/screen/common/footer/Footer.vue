<template>
  <div
    class="flex items-center justify-between px-8 py-5 max-w-md relative"
    style="background-color: #3b3b3b"
  >
    <div
      class="absolute text-white rounded-full bg-green-500 cursor-pointer"
      style="top: 0; left: 50%; transform: translate(-50%, -50%)"
    >
      <img
        class="m-3 hover:opacity-75"
        src="@/assets/icons/add.svg"
        width="20"
        height="20"
        alt="add"
      />
    </div>
    <img
      class="cursor-pointer hover:opacity-75"
      :class="active('grid')"
      src="@/assets/icons/grid.svg"
      width="25"
      height="25"
      alt="grid"
      @click="clickListener('grid')"
    />
    <img
      class="cursor-pointer hover:opacity-75"
      :class="active('dumbbell')"
      src="@/assets/icons/dumbbell.svg"
      width="25"
      height="25"
      alt="dumbbell"
      @click="clickListener('dumbbell')"
    />
    <img
      class="cursor-pointer hover:opacity-75"
      :class="active('graph')"
      src="@/assets/icons/graph.svg"
      width="25"
      height="25"
      alt="graph"
      @click="clickListener('graph')"
    />
    <img
      class="cursor-pointer hover:opacity-75"
      :class="active('calendar')"
      src="@/assets/icons/calendar.svg"
      width="25"
      height="25"
      alt="calendar"
      @click="clickListener('calendar')"
    />
  </div>
</template>
<script>
import { defineComponent, ref, computed } from '@nuxtjs/composition-api'

/**
 * #Footerコンポーネント(SP幅専用)
 */
export default defineComponent({
  setup() {
    const current = ref('grid')
    const active = computed(() => (val) =>
      current.value === val ? 'opacity-100' : 'opacity-25'
    )

    function clickListener(val) {
      current.value = val
    }

    return {
      clickListener,
      active,
    }
  },
})
</script>
