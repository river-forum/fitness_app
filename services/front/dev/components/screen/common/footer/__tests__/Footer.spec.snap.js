import { mount } from '@vue/test-utils'
import Footer from '../Footer.vue'

it('Footer Component renders correctly', () => {
  const app = mount(Footer)

  expect(app.html()).toMatchSnapshot()
})
