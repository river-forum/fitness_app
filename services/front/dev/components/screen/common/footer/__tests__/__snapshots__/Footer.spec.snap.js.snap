// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component renders correctly 1`] = `
"<div class=\\"flex items-center justify-between px-8 py-5 max-w-md relative\\" style=\\"background-color: rgb(59, 59, 59);\\">
  <div class=\\"absolute text-white rounded-full bg-green-500 cursor-pointer\\" style=\\"top: 0px; left: 50%; transform: translate(-50%, -50%);\\"><img src=\\"~assets/icons/add.svg\\" width=\\"20\\" height=\\"20\\" alt=\\"add\\" class=\\"m-3 hover:opacity-75\\"></div> <img src=\\"~assets/icons/grid.svg\\" width=\\"25\\" height=\\"25\\" alt=\\"grid\\" class=\\"cursor-pointer hover:opacity-75 opacity-100\\"> <img src=\\"~assets/icons/dumbbell.svg\\" width=\\"25\\" height=\\"25\\" alt=\\"dumbbell\\" class=\\"cursor-pointer hover:opacity-75 opacity-25\\"> <img src=\\"~assets/icons/graph.svg\\" width=\\"25\\" height=\\"25\\" alt=\\"graph\\" class=\\"cursor-pointer hover:opacity-75 opacity-25\\"> <img src=\\"~assets/icons/calendar.svg\\" width=\\"25\\" height=\\"25\\" alt=\\"calendar\\" class=\\"cursor-pointer hover:opacity-75 opacity-25\\">
</div>"
`;
